app/models/user.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+# User model specifications
+class User < ApplicationRecord
+  # formats the email to down case before saving
+  attr_accessor :remember_token
+  # attr_accessor :token
+
+  before_create :first_token
+  before_save { self.email = email.downcase }
+
+  # Allows email validations for correct format
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i.freeze
+
+  validates :username, presence: true, length: { minimum: 3, maximum: 50 }
+  validates :password, presence: true, length: { minimum: 6, maximum: 255 }
+  validates :email, presence: true, length: { minimum: 10, maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+
+  has_secure_password
+  has_many :posts
+
+  class << self
+    # Returns the hash digest of the given string.
+    def digest(string)
+      Digest::SHA1.hexdigest(string.to_s)
+    end
+
+    # Returns a random token.
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
```

Comment:
For that project, you can merge them into one method so that it return  the digest of a token directly.

Reply:
## Refactored

```
  def create_remember_token
    rem_token = Digest::SHA1.hexdigest(SecureRandom.urlsafe_base64)
    self.remember_token = rem_token
  end 
```